lib/std/rand.zig

```diff
@@ -776,6 +776,22 @@ pub const Gimli = struct {
     }
 };
 
+/// Cryptographically secure random number generator based on system calls.
+/// This is just a wrapper for `std.crypto.randomBytes()` that implements the `Random` interface.
+pub const OsRandom = struct {
+    random: Random,
+
+    pub fn init() OsRandom {
+        return OsRandom{
+            .random = Random{ .fillFn = fill },
+        };
+    }
```

Comment:
proposal: delete this function, and follow this pattern: 

https://github.com/ziglang/zig/blob/1e13e8e8172c1409c5152e8b4e935c5a1d31704b/lib/std/heap.zig#L88-L98

Combined with moving it to std.crypto, usage would look like this:

```zig
std.crypto.os_random.bytes(buf);
```

Which I think would be enough to justify deleting `std.crypto.randomBytes`.  This would be a good time to plan what the path to #6704 would be. Perhaps `std.crypto.random` meaning that we could tell users to upgrade from `std.crypto.randomBytes` to `std.crypto.random.bytes` which is a pretty satisfying upgrade, considering that `bytes` could then be replaced with any of the other Random interface functions, and it gains the benefits of the thread-local CSPRNG infrastructure.